scripts/extensions/ai-widget/src/ai-assistant.tsx

```diff
@@ -77,43 +188,69 @@ export class AiAssistantWidget extends React.PureComponent<IArticleSideWidgetCom
         const closeActiveSection = () => {
             this.setState({activeSection: null});
         };
-        const headlinesWidget = getHeadlinesWidget({
-            closeActiveSection,
-            article: this.props.article,
-            error: this.state.error,
-            generateHeadlines: this.generateHeadlines,
-            headlines: this.state.headlines,
-            loading: this.state.loadingHeadlines,
-            reGenerateHeadlines: () => {
-                this.setState({
-                    loadingHeadlines: true,
-                }, () => this.generateHeadlines());
-            },
-            fieldsData: this.props.fieldsData,
-            onFieldsDataChange: this.props.onFieldsDataChange,
-        });
-        const summaryWidget = getSummaryWidget({
-            closeActiveSection,
-            article: this.props.article,
-            error: this.state.error,
-            generateSummary: this.generateSummary,
-            summary: this.state.summary,
-            loading: this.state.loadingSummary,
-            regenerateSummary: () => {
-                this.setState({
-                    loadingSummary: true,
-                }, () => this.generateSummary());
-            },
-        });
+        const state = this.state;
+
         const currentComponent: {
             header?: JSX.Element;
             body: JSX.Element;
             footer?: JSX.Element;
         } = (() => {
-            if (this.state.activeSection === 'headlines') {
-                return headlinesWidget;
-            } else if (this.state.activeSection === 'summary') {
-                return summaryWidget;
+            if (state.activeSection === 'headlines') {
```

Comment:
move widget specific functions like `generateTranslations` inside specific widgets(translation widget in this case) and your currentComponent should look something like this:

```
interface ICommonProps extends IArticleSideWidgetComponentType {
    state: IState;
    onStateChange(state: IState): void;
}

const componentsByTab: {
    [KEY in NonNullable<IState['activeSection']>]: React.ComponentType<ICommonProps>;
} = {
    'headlines': getHeadlinesWidget,
    'summary': getSummaryWidget,
    'translations': getTranslationsWidget,
};

const currentComponent = componentsByTab[state.activeSection];
```

Result is, reduced amount of passed props that clutter the code and also unified props for all components.